### ISSUE TITLE
Istio Proxy Memory Leak

### ISSUE THREAD

**Issue by crhuber** [istio/istio#37014]
### Bug Description

Hello,
We have some istio proxies running 1.11.3 that have a large memory growth over time but is not limited to version 1.11.x

For example in 2 days from ~50MB to 80MB
![image](https://user-images.githubusercontent.com/5857391/151197889-0941df5d-402a-4ce4-9886-befaee4da6a7.png).
The application container is nginx and is configured with these settings:

```
keepalive_timeout  65;
keepalive_requests 100;
```
On a longer time line 26 weeks, I can see memory use balloons to 400mb sometimes
`sum(container_memory_usage_bytes{namespace="menu",pod=~"service-api.*", container="istio-proxy"} / 1024^2) by (pod, container) `
![image](https://user-images.githubusercontent.com/5857391/151200151-bbe2565e-1168-4d37-b3d3-70ec98f48ebb.png)

otherwise this service has the following destination rule:

```
apiVersion: networking.istio.io/v1beta1
kind: DestinationRule
...
  trafficPolicy:
    connectionPool:
      http:
        http1MaxPendingRequests: 1024
        maxRequestsPerConnection: 1024
      tcp:
        maxConnections: 1024
```
It also has a `Sidecar` resource to reduce the envoy cluster footprint

```
apiVersion: networking.istio.io/v1beta1
kind: Sidecar
spec:
  egress:
  - hosts:
    - istio-system/*
  workloadSelector:
    labels:
      app:service-api
```

cluster config
```
istioctl1114 proxy-config clusters service-api-65fdc47646-4sfrb
SERVICE FQDN                                                PORT      SUBSET     DIRECTION     TYPE             DESTINATION RULE
                                                            80        -          inbound       ORIGINAL_DST
BlackHoleCluster                                            -         -          -             STATIC
InboundPassthroughClusterIpv4                               -         -          -             ORIGINAL_DST
InboundPassthroughClusterIpv6                               -         -          -             ORIGINAL_DST
PassthroughCluster                                          -         -          -             ORIGINAL_DST
agent                                                       -         -          -             STATIC
istio-ingressgateway-tcp.istio-system.svc.cluster.local     80        -          outbound      EDS
istio-ingressgateway-tcp.istio-system.svc.cluster.local     443       -          outbound      EDS
istio-ingressgateway-tcp.istio-system.svc.cluster.local     15021     -          outbound      EDS
istio-ingressgateway.istio-system.svc.cluster.local         80        -          outbound      EDS
istio-ingressgateway.istio-system.svc.cluster.local         443       -          outbound      EDS
istio-ingressgateway.istio-system.svc.cluster.local         15021     -          outbound      EDS
istiod-istio1114.istio-system.svc.cluster.local             443       -          outbound      EDS
istiod-istio1114.istio-system.svc.cluster.local             15010     -          outbound      EDS
istiod-istio1114.istio-system.svc.cluster.local             15012     -          outbound      EDS
istiod-istio1114.istio-system.svc.cluster.local             15014     -          outbound      EDS
prometheus_stats                                            -         -          -             STATIC
sds-grpc                                                    -         -          -             STATIC
xds-grpc                                                    -         -          -             STATIC
zipkin                                                      -         -          -             STRICT_DNS
```

ss stats (after 2 days)
```
ss -s
Total: 237
TCP:   3691 (estab 217, closed 3465, orphaned 2, timewait 771)

Transport Total     IP        IPv6
RAW	  0         0         0
UDP	  1         1         0
TCP	  226       220       6
INET	  227       221       6
FRAG	  0         0         0
```

When I find the number of time series after 2 days for istio_requests I get `61`

```
curl localhost:15000/stats/prometheus | grep istio_requests_total | nl
61
```

We do not have coreDump enabled ( values.global.proxy.enableCoreDump=true)  meshwide to do a pprof as describe here
https://github.com/istio/istio/wiki/Analyzing-Istio-Performance

### Version

```prose
client version: 1.11.4
control plane version: 1.11.4
data plane version: 1.11.4 (3117 proxies)
kube
Client Version: v1.22.2
Server Version: v1.21.5-eks-bc4871b
```

**Comment by crhuber** [comment]
This seems to have resolved itself in Istio 1.13.2

**Comment by Umenezumi** [comment]
> This seems to have resolved itself in Istio 1.13.2

How did you fix it?
